tests/filtered_relation/tests.py

```diff
@@ -0,0 +1,151 @@
+import unittest
+
+from django.core.exceptions import FieldError
+from django.db import connection
+from django.db.models import Q
+from django.test import TestCase
+
+from .models import Author, Book, Editor
+
+
+class FilteredRelationTests(TestCase):
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.author1 = Author.objects.create(name='Alice')
+        cls.author2 = Author.objects.create(name='Jane')
+
+        cls.editor_a = Editor.objects.create(name='a')
+        cls.editor_b = Editor.objects.create(name='b')
+
+        cls.book1 = Book.objects.create(title='Poem by Alice',
```

Comment:
This file isn't following the indentation style of other tests. Please use hanging 4 space indent.